src/tasks/WasmAppBuilder/WasmAppBuilder.csproj

```
@@ -13,13 +22,32 @@
   </ItemGroup>
 
   <Target Name="PublishBuilder"
-          AfterTargets="Build"
-          DependsOnTargets="Publish" />
+          AfterTargets="Build">
+
+    <!-- needed for publishing with multi-targeting. We are publishing essentially to get the SR.MetadataLoadContext.dll :/ -->
+    <ItemGroup>
+      <_PublishFramework Include="$(TargetFrameworks)" />
+    </ItemGroup>
+    <MSBuild Projects="$(MSBuildProjectFile)" Targets="Publish" Properties="TargetFramework=%(_PublishFramework.Identity)" />
+  </Target>
 
   <Target Name="GetFilesToPackage" Returns="@(FilesToPackage)">
     <ItemGroup>
-      <FilesToPackage Include="$(OutputPath)$(MSBuildProjectName)*" TargetPath="tasks" />
-      <FilesToPackage Include="$(PublishDir)System.Reflection.MetadataLoadContext.dll" TargetPath="tasks" />
+      <_PublishFramework Remove="@(_PublishFramework)" />
+      <_PublishFramework Include="$(TargetFrameworks)" />
+
+      <!-- non-net4* -->
+      <FilesToPackage Include="$(OutputPath)%(_PublishFramework.Identity)\$(MSBuildProjectName)*"
+                      Condition="!$([System.String]::Copy('%(_PublishFramework.Identity)').StartsWith('net4'))"
+                      TargetPath="tasks\%(_PublishFramework.Identity)" />
+      <FilesToPackage Include="$(OutputPath)%(_PublishFramework.Identity)\publish\System.Reflection.MetadataLoadContext.dll"
+                      Condition="!$([System.String]::Copy('%(_PublishFramework.Identity)').StartsWith('net4'))"
+                      TargetPath="tasks\%(_PublishFramework.Identity)" />
```

I don't think batching over the publish frameworks is necessary here.  There are only two of them, and you have conditions which apply separate logic to each one.  Would something like the following work?

```suggestion
      <FilesToPackage Include="$(OutputPath)$(NetCoreAppToolCurrent)\$(MSBuildProjectName)*"
                      TargetPath="tasks\$(NetCoreAppToolCurrent)" />
      <FilesToPackage Include="$(OutputPath)$(NetCoreAppToolCurrent)\publish\System.Reflection.MetadataLoadContext.dll"
                      TargetPath="tasks\$(NetCoreAppToolCurrent)" />
```

---

Yeah, this should be fine too. I was trying to ensure that this doesn't break if we only change `$(TargetFrameworks)`, and not update this.